graphql_compiler/tests/test_compiler.py

```diff
@@ -5907,16 +5939,16 @@ def test_multiple_filters_on_count(self):
                 GROUP BY "Animal_2".uuid
             ) AS folded_subquery_1
             ON "Animal_1".uuid = folded_subquery_1.uuid
-            LEFT OUTER JOIN (
+            JOIN (
                 SELECT
-                    "Animal_4".related_entity AS related_entity,
+                    "Animal_4".uuid AS uuid,
```

Comment:
Similar as comment above. This was introduced in #621 but this PR changed to using primary keys for the subquery